src/library/Table/TableSortableHeaderCell.js

```diff
@@ -137,62 +148,93 @@ const sortIcon = {
   descending: <IconArrowDropdownDown {...iconProps} />
 };
 
-export default function TableSortableHeaderCell({
-  children,
-  label,
-  name,
-  messages,
-  sortable,
-  ...restProps
-}: Props) {
-  const { sort, sortFn } = sortable;
-
-  const sortColumn = sort && sort.key;
-  const descending = sort ? sort.descending : false;
-
-  const isActiveSort = sortColumn === name;
-  const activeDirection = descending ? 'descending' : 'ascending';
-  const nextDirection =
-    isActiveSort && activeDirection === 'ascending'
-      ? 'descending'
-      : 'ascending';
-
-  const toggle = isActiveSort || activeDirection !== nextDirection;
-  const nextDescending = toggle ? !descending : descending;
-
-  const a11yLabel = label || children;
-
-  const rootProps = {
-    ...restProps,
-    'aria-label': a11yLabel,
-    'aria-sort': isActiveSort ? activeDirection : 'none',
-    role: 'columnheader'
-  };
-
-  const buttonProps = {
-    ...restProps,
-    'aria-label':
-      nextDirection === 'ascending'
-        ? messages.sortColumnAscending
-        : messages.sortColumnDescending,
-    onClick: () => {
-      sortFn({ key: name, descending: nextDescending });
-    }
-  };
-  const iconHolderProps = {
-    direction: (isActiveSort && activeDirection) || 'ascending',
-    isActiveSort
-  };
-
-  return (
-    <Root {...rootProps}>
-      <Button {...buttonProps}>
-        <Content>{children}</Content>
-        &nbsp;
-        <IconHolder {...iconHolderProps}>
-          {sortIcon[iconHolderProps.direction]}
-        </IconHolder>
-      </Button>
-    </Root>
-  );
+/**
+ * TableSortableHeaderCell
+ */
+export default class TableSortableHeaderCell extends Component<Props> {
+  render() {
+    countRender('TableSortableHeaderCell');
+
+    return (
+      <TableContext.Consumer>
+        {(tableContextProps) => {
+          const {
+            children,
+            label,
+            name,
+            messages,
+            render,
+            sortable,
+            ...restProps
+          } = this.props;
+          const { sort, sortFn } = sortable;
+          const sortColumn = sort && sort.key;
+          const descending = sort ? sort.descending : false;
+          const direction = descending ? 'descending' : 'ascending';
+          const isSorted = sortColumn === name;
+          const nextDirection =
+            isSorted && direction === 'ascending' ? 'descending' : 'ascending';
+          const toggle = isSorted || direction !== nextDirection;
+          const nextDescending = toggle ? !descending : descending;
+          const nextSort = { key: name, descending: nextDescending };
+          const handleSort = () => sortFn(nextSort);
+
+          const rootProps = {
+            'aria-label': label || children,
+            'aria-sort': isSorted ? direction : 'none',
+            role: 'columnheader',
+            ...restProps
```

Comment:
Do you remember why we used to spread on top and now spread on bottom?

Reply:
I don't know why it was spread on top previously.  That is generally a bad pattern as it does not allow overrides.  The only thing that I could think would be to enforce that those keys don't get overridden.